feat(trainings): remove broken trainings and replace some with blog style training (#3353)
patternfly/patternfly-org: packages/v4/patternfly-docs/content/training/html-css/css-variables-and-overrides.md

PatternFly uses a modified version of BEM for its CSS architecture. Where PatternFly deviates from BEM is in relation to modifiers.

This tutorial covers how PatternFly uses BEM as a framework for its component library. You'll learn how to override and create both global and component-level custom properties and component elements.

## Part 1: Overriding PatternFly component-level properties
In PatternFly, component-level custom properties follow this general formula:
`--pf-c-block[__element][--modifier][--state][--breakpoint][--pseudo-element]--PropertyCamelCase.`
 - `--pf-c-block` refers to the block, usually the component or layout name (e.g. `--pf-c-alert`).
 - `__element` refers to the element inside of the block (e.g. `__title`).
 - `--modifier` refers to a modifier class such as `.pf-m-danger`, and is prefixed with `m-` in the component variable (e.g.`--m-danger`).
 - `--state` is something like hover or active.
 - `--breakpoint` is a media query breakpoint such as sm for `$pf-global--breakpoint--sm`.
 - `--pseudo-element` is one of either before or after.
 - `PropertyCamelCase` refers to the property that is being changed.

> To explore this concept, you can practice overriding the title color custom property in the success variation of the alert component using <a href="https://codesandbox.io/s/html-fundamentals-start-hy2erg" target="_blank">this codesandbox for part 1</a>.

Reviewer: I'm not seeing the alert component in this sandbox

The resulting alert should match Figure 1.

**Figure 1**
<img src="../img/alert-red-title.png" alt="Alert component with a red title" width="868" />

## Part 2: Overriding PatternFly global properties
In PatternFly, global properties follow this general formula:
`--pf-global--concept--PropertyCamelCase--modifier--state`

Global properties are prefixed with the word global.
 - A `concept` is something like a spacer or main-title.
 - A `PropertyCamelCase` is something like `BackgroundColor` or `FontSize`.
 - A `modifier` is something like sm, or lg.
 - A `state` is something like hover, or expanded. They are concepts, never tied to an element or component. This is incorrect: `--pf-global--h1--FontSize`. This is correct: `--pf-global--FontSize--3xl`.

To explore this concept, override the global link color by setting it to the custom property for danger using <a href="https://codesandbox.io/s/override-global-css-variables-start-llcub8" target="_blank">this codesandbox for part 2</a>. The `index.html` file contains a button and expandable component to demonstrate how changing a global variable has an impact across components.

### Step 1. Familiarize with `index.html`

> Note the two button component and the expandable section component in the `<body>` of the `index.html` file.

Reviewer: "two button components" - or I actually only see one button?

#### Step 2.1 
Write the prefix for the global custom property in the `:root` block in the `styles.css` file. Since global custom properties are prefixed with the word global, use that for the beginning of the custom property.
It should look like this: `--pf-global`

#### Step 2.2 
Add the concept to the global custom property. The concept for the link color is link.
The global custom property should now look like this: `--pf-global--link`

#### Step 2.3 
Add the property to the global custom property. As the color is being modified, add the color property as the next step.
The global custom property should now look like: `--pf-global--link--Color`

#### Step 2.4 
Define the value of the global variable. PatternFly's global danger color is: `--pf-global--danger-color--100`.
Assign the custom property name `(--pf-global--link--Color)` that is already inside of the `:root` block to the global danger color.

It should look like this:
`--pf-global--link--Color: var(--pf-global--danger-color--100);`

The resulting UI in the codesandbox should match Figure 2. **Note:** The button component and expandable component should inherit different colors.

**Figure 2.**

> <img src="../img/global-link-color-red.png" alt="An inline button and a 'show more' expandable section toggle both colored red" width="200" />

Reviewer: The previous line says they should inherit different colors, but this shows they are both red? (and they are when I do it)

---
id: HTML CSS variables and overrides training
section: training
hideNavItem: true
---
import { Button, ClipboardCopy } from '@patternfly/react-core';

# CSS variables and overrides

> PatternFly Core is based on the principles of <a href="http://bradfrost.com/blog/post/atomic-web-design/" target="_blank">Atomic design</a> and BEM (Block, Element, Modifier). <a href="http://getbem.com/introduction/" target="_blank">BEM</a> is a popular CSS methodology for building modular, scalable applications. It provides scope, avoids inheritance, and reduces CSS specificity and conflicts.

Reviewer 2: ```suggestion
PatternFly Core is based on the principles of <a href="http://bradfrost.com/blog/post/atomic-web-design/" target="_blank">Atomic Design</a> and <a href="http://getbem.com/introduction/" target="_blank">BEM</a> Block, Element, Modifier). BEM is a popular CSS methodology for building modular, scalable applications. It provides scope, avoids inheritance, and reduces CSS specificity and conflicts.
```

small technicalities - it seems like Atomic Design refers to a book & is a proper noun, so it should be capitalized. I would also go ahead and link to BEM the first time it's mentioned.

BEM works perfectly with a modular design system, as each unique component can be represented as an independent block. Since a block is tied to a component, developers are able to develop, move around, and nest components without conflicts in their application’s CSS.

> PatternFly uses a modified version of BEM for its CSS architecture. Where PatternFly deviates from BEM is in relation to modifiers.

Reviewer 2: ```suggestion
PatternFly uses a modified version of BEM for its CSS architecture. PatternFly deviates from BEM in relation to modifiers.
```

This tutorial covers how PatternFly uses BEM as a framework for its component library. You'll learn how to override and create both global and component-level custom properties and component elements.

## Part 1: Overriding PatternFly component-level properties
In PatternFly, component-level custom properties follow this general formula:
`--pf-c-block[__element][--modifier][--state][--breakpoint][--pseudo-element]--PropertyCamelCase.`

>  - `--pf-c-block` refers to the block, usually the component or layout name (e.g. `--pf-c-alert`).

Reviewer 2: ```suggestion
 - `--pf-c-block` refers to the block, usually the component or layout name (for example,`--pf-c-alert`).
```

PF suggests avoiding "e.g." ([here](https://www.patternfly.org/v4/ux-writing/terminology#abbreviations-and-acronyms)), so just tweaking

>  - `__element` refers to the element inside of the block (e.g. `__title`).

Reviewer 2: ```suggestion
 - `__element` refers to the element inside of the block (for example, `__title`).
```

>  - `--modifier` refers to a modifier class such as `.pf-m-danger`, and is prefixed with `m-` in the component variable (e.g.`--m-danger`).

Reviewer 2: ```suggestion
 - `--modifier` refers to a modifier class such as `.pf-m-danger`, and is prefixed with `m-` in the component variable (for example,`--m-danger`).
```

 - `--state` is something like hover or active.

>  - `--breakpoint` is a media query breakpoint such as sm for `$pf-global--breakpoint--sm`.

Reviewer 2: ```suggestion
 - `--breakpoint` is a media query breakpoint such as `sm` for `$pf-global--breakpoint--sm`.
```

<img src="../img/alert-red-title.png" alt="Alert component with a red title" width="868" />

## Part 2: Overriding PatternFly global properties
In PatternFly, global properties follow this general formula:
`--pf-global--concept--PropertyCamelCase--modifier--state`

Global properties are prefixed with the word global.
 - A `concept` is something like a spacer or main-title.
 - A `PropertyCamelCase` is something like `BackgroundColor` or `FontSize`.
 - A `modifier` is something like sm, or lg.
 - A `state` is something like hover, or expanded. They are concepts, never tied to an element or component. This is incorrect: `--pf-global--h1--FontSize`. This is correct: `--pf-global--FontSize--3xl`.

To explore this concept, override the global link color by setting it to the custom property for danger using the provided codesandbox for part 2. The `index.html` file contains a button and expandable component to demonstrate how changing a global variable has an impact across components.

<Button variant="primary" component="a" href="https://codesandbox.io/s/override-global-css-variables-start-llcub8" target="_blank">Codesandbox - Part 2</Button>

### Step 1. Familiarize with `index.html`

> Note of the button component and the expandable section component in the `<body>` of the `index.html` file.

Reviewer 2: ```suggestion
Note the button component and the expandable section component in the `<body>` of the `index.html` file.
```

### Step 2. Override a global custom property

#### Step 2.1 
Write the prefix for the global custom property in the `:root` block in the `styles.css` file. Since global custom properties are prefixed with the word global, use that for the beginning of the custom property.

It should look like this: `--pf-global`

#### Step 2.2 

> Add the concept to the global custom property. The concept for the link color is link.

Reviewer 2: ```suggestion
Add the concept to the global custom property. The concept for the link color is `link`.
```
Not sure if it'd be better to use backticks or quotation marks in the places where I'm doing so. Mostly want to help with the readability & separate things that are intended to be user-entered values.
